linker/Mono.Linker.Steps/ResolveFromAssemblyStep.cs

```diff
@@ -67,7 +67,7 @@ static void SetAction (LinkContext context, AssemblyDefinition assembly, Assembl
 		{
 			TryReadSymbols (context, assembly);
 
-			context.Annotations.SetAction (assembly, action);
+			context.SetAction (assembly, action);
 		}
```

Comment:
context.SetAction () knows about the per-assembly actions, context.Annotations.SetAction () doesn't.